plugins/speculation-rules/hooks.php

```diff
@@ -58,3 +58,75 @@ function plsr_render_generator_meta_tag(): void {
 	echo '<meta name="generator" content="speculation-rules ' . esc_attr( SPECULATION_RULES_VERSION ) . '">' . "\n";
 }
 add_action( 'wp_head', 'plsr_render_generator_meta_tag' );
+
+/**
+ * Filters the HTML output of the search form to inject speculative loading interactivity.
+ *
+ * @since n.e.x.t
+ *
+ * @param string|mixed $form The search form HTML output.
+ * @return string Filtered HTML.
+ */
+function plsr_filter_searchform( $form ): string {
+	if ( ! is_string( $form ) ) {
+		return '';
+	}
+
+	$namespace        = 'speculationRules';
+	$directive_prefix = 'data-wp-on'; // TODO: Use data-wp-on-async when available.
+
+	$p = new WP_HTML_Tag_Processor( $form );
+	while ( $p->next_tag() ) {
+		if ( 'FORM' === $p->get_tag() ) {
+			if ( ! $p->get_attribute( 'data-wp-interactive' ) ) {
+				$p->set_attribute( 'data-wp-interactive', $namespace );
+			}
+			// Create context if not already present.
+			// TODO: Should there be namespaced context?
+			if ( ! $p->get_attribute( 'data-wp-context' ) ) {
+				$p->set_attribute( 'data-wp-context', '{}' );
+			}
```

Comment:
See `wp_interactivity_data_wp_context()` which has a `$store_namepace` argument. The logic in that function could be replicated here. However, it seems this would conflict when there is already a `data-wp-context` attribute present.

Reply:
I have created a discussion to see how we can solve that problem in a general way for all the directives:
- https://github.com/WordPress/gutenberg/discussions/62720